docs/modules/data-structures/pages/vector-search-tutorial.adoc

```diff
@@ -0,0 +1,183 @@
+
+. Test the pipeline
++
+.. build the project: `mvn clean package`
+.. deploy the pipeline: `docker compose run deploy-image-loader`
+.. monitor the logs: `docker compose logs --follow hz`
+.. to generate events, copy some images from the `images` folder
+into the `www` folder.  You will see a new pipeline event for each image.
+.. To stop the job, use the Hazelcast Management Center. Navigate
+to "Jobs" and select "Stop".
++
+Iterate on the pipeline until you have finished the whole thing, and you are sinking vectors into Hazelcast with no errors.
++
+Once you have the python service in your pipeline, it will take some time to initialize (up to 5 minutes) because it has to download and install many Python packages.  You will see something like the following in the hazelcast logs when the Python stream stage has initialized.
+
+```bash 
+hazelcast-image-search-hz-1  | 2024-07-17 19:18:41,881 [ INFO] [hz.magical_joliot.cached.thread-7] [c.h.j.python]: [172.25.0.3]:5701 [dev] [5.5.0-SNAPSHOT] Started Python process: 246
+hazelcast-image-search-hz-1  | 2024-07-17 19:18:41,881 [ INFO] [hz.magical_joliot.cached.thread-3] [c.h.j.python]: [172.25.0.3]:5701 [dev] [5.5.0-SNAPSHOT] Started Python process: 245
+hazelcast-image-search-hz-1  | 2024-07-17 19:18:43,786 [ INFO] [hz.magical_joliot.cached.thread-7] [c.h.j.python]: [172.25.0.3]:5701 [dev] [5.5.0-SNAPSHOT] Python process 246 listening on port 39819
+hazelcast-image-search-hz-1  | 2024-07-17 19:18:43,819 [ INFO] [hz.magical_joliot.cached.thread-3] [c.h.j.python]: [172.25.0.3]:5701 [dev] [5.5.0-SNAPSHOT] Python process 245 listening on port 39459
+```
+
+[NOTE]
+====
+A solution pipeline is available in the
+`hazelcast.platform.labs.image.similarity.solution` package.  If you wish, you can bypass building the pipeline and directly deploy the solution by running
+`docker compose run submit-image-loader-solution`
+====
+
+== Perform a Nearest Neighbor Search
+
+You will do the rest of the lab in a Jupyter notebook.  To access Jupyter,
+run `docker compose logs jupyter`.  You will see something simular to
+what is below.
+
+```bash
+hazelcast-image-search-jupyter-1  | [C 2024-07-17 19:57:47.478 ServerApp]
+hazelcast-image-search-jupyter-1  |
+hazelcast-image-search-jupyter-1  |     To access the server, open this file in a browser:
+hazelcast-image-search-jupyter-1  |         file:///root/.local/share/jupyter/runtime/jpserver-1-open.html
+hazelcast-image-search-jupyter-1  |     Or copy and paste one of these URLs:
+hazelcast-image-search-jupyter-1  |         http://localhost:8888/tree?token=7a4d2794d4135eaa88ee9e9642e80e7044cb5c213717e2be
+hazelcast-image-search-jupyter-1  |         http://127.0.0.1:8888/tree?token=7a4d2794d4135eaa88ee9e9642e80e7044cb5c213717e2be
+haz
+```
+
+[NOTE]
+====
+Some errors in the log are expected and can be safely ignored.  The lines you are looking for should be near the top of the output.
```

Comment:
Which errors?
Is it worth highlighting that user should be looking for the URL?